compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```
@@ -451,8 +451,33 @@ getMatmulLoweringConfigAndWorkgroupSize(SmallVector<int64_t> bounds,
       kPackFactor = std::get<2>(mmaKind.getMNKShape());
     }
     paddingTileSizes[innerKDim] *= kPackFactor;
-    attrs.emplace_back(StringAttr::get(context, "padding"),
-                       b.getI64ArrayAttr(paddingTileSizes));
+
+    // In the case of padding convs, starting from workgroup tile sizes, only
+    // the reduction dimensions require appropriate padding.
```

This only works because we don't collapse any parallel dimensions when doing IGEMM. If we eventually change this, then this will silently succeed, and create incorrect padding, which could be hard to track down. Could you add a check here that the number of `M` dimensions in the GEMM iteration space matches the number of `outputImage` + `batch` dimensions of the convolution, and that the number of `N` dimensions of the GEMM matches the number of `outputChannel` dimensions of the conv, and fallback to regular padding if it does not match?

---

Actually, this also should check that the actual dimensions match too, to be safe. In other words, the M dimensions of the matmul should be equal to the concatenated batch and outputImage dims, so if the convolution has `batch=[0]` and `outputImage=[1, 2]`, then the matmul should have `M=[0, 1, 2]`. The `N` dims should match the `outputChannel` dims exactly too.

---

I think the above only works for NHWC layout? For NCHW, the M and N dimension is swapped. Is it better to assign each dimension of the convolution padding size with the corresponding M/N dimension tile sizes? Or do you think it's okay to use the order of `workgroupTileSizes` and just check the corresponding dimension sizes? The former is a bit more complicated than the latter consider different convolution layouts.

---

For now, just use the order of `workgroupTileSizes` and verify the dimensions match. I think this strategy won't work well for NCHW convs anyway, so we only really care about NHWC right now. In the future, the iteration spaces of the GEMM and conv might not match, though, at which point we probably want to refactor the config setting functions anyway, so let's do the simple thing for now.

---

Actually I ended up refactoring the codes to assign values for parallel dimensions as well.